bitcoin/src/crypto/key.rs

```diff
@@ -507,17 +507,27 @@ impl fmt::Display for TweakedPublicKey {
 }
 
 /// Untweaked BIP-340 key pair
-pub type UntweakedKeyPair = KeyPair;
+#[deprecated(since = "0.31.0", note = "use UntweakedKeypair instead")]
+#[allow(deprecated)]
+pub type UntweakedKeyPair = UntweakedKeypair;
+
+/// Untweaked BIP-340 key pair
+pub type UntweakedKeypair = Keypair;
+
+/// Tweaked BIP-340 key pair
+#[deprecated(since = "0.31.0", note = "use TweakedKeypair instead")]
+#[allow(deprecated)]
+pub type TweakedKeyPair = TweakedKeypair;
```

Comment:
My 2sats here from a person that does not like type alias are: what is the reason for re-aliasing this type?

Rust docs do not do a really good job of putting documentation on the re-alias type so due to this is not doing really anything it is possible to re-evaluate the type alias here?

Reply:
We are trying to minimize or eliminate this. Sometimes it's not possible. But in the case of pure renames of structs or functions, it definitely is possible :).

Reply:
Thanks for the review @vincenzopalazzo, appreciate your input. I had play with this and rendered the docs with various comments on it. The alias does show up with a link to the original type and the deprecation is handled in a clear way also. I think this is ok.

FTR we remove deprecated stuff in the next release or two so this won't be here for long. (I _think_ our policy is deprecate for one release cycle but I may be wrong, it might be two?).